tools/check.py

```diff
@@ -667,6 +667,42 @@ def command_line():
 
     return parser.parse_args()
 
+def check_required_files(files):
+    """Check if required files exists."""
+    REQUIRED_FILES = ["..-.*\.md",
+                      "discussion.md",
+                      "index.md",
+                      "instructors.md",
+                      "LICENSE.md",
+                      "motivation.md",
+                      "README.md",
+                      "reference.md"]
+    valid = True
+
+    for required in REQUIRED_FILES:
+        if not [True for fn in files if re.search(required, fn) is not None]:
```

Comment:
Using os.path.exists is preferable to regex-matching filenames, because:

a) It may not correctly enforce expected folder structure, especially if the validator is run from somewhere other than the top level folder.

b) Windows is case-insensitive; straight string comparison of filenames is not platform-independent. (and can generally lead to weird behavior)